From 7e57ff280b55b45e74329b9988279e8831d32eab Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Sun, 25 Jan 2015 09:45:04 +0100
Subject: [PATCH] cmake: use the standard CMake flag to drive the shared
 object build

Remove the STATICLIBS CMake option (and the code handling it) and let
the standard CMake flags drive the shared object build.

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[Fabrice: update for 1.5]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 CMakeOptions.txt      |  1 -
 ftdipp/CMakeLists.txt | 14 +-------------
 src/CMakeLists.txt    | 13 +------------
 3 files changed, 2 insertions(+), 26 deletions(-)

diff --git a/CMakeOptions.txt b/CMakeOptions.txt
index 07b5887..b780ffd 100644
--- a/CMakeOptions.txt
+++ b/CMakeOptions.txt
@@ -1,4 +1,3 @@
-option ( STATICLIBS "Build static libraries" ON )
 option ( BUILD_TESTS "Build unit tests with Boost Unit Test framework" OFF )
 option ( DOCUMENTATION "Generate API documentation with Doxygen" OFF )
 option ( EXAMPLES "Build example programs" ON )
diff --git a/ftdipp/CMakeLists.txt b/ftdipp/CMakeLists.txt
index fac5bcc..a06edf1 100644
--- a/ftdipp/CMakeLists.txt
+++ b/ftdipp/CMakeLists.txt
@@ -12,7 +12,7 @@ include_directories(BEFORE ${CMAKE_CURRENT_BINARY_DIR}
 include_directories(${Boost_INCLUDE_DIRS})
 
 # Shared library
-add_library(ftdipp1 SHARED ${cpp_sources})
+add_library(ftdipp1 ${cpp_sources})
 
 math(EXPR VERSION_FIXUP "${MAJOR_VERSION} + 1")    # Compatibility with previous releases
 set_target_properties(ftdipp1 PROPERTIES VERSION ${VERSION_FIXUP}.${MINOR_VERSION}.0 SOVERSION 3)
@@ -29,18 +29,6 @@ install ( TARGETS ftdipp1
 					ARCHIVE DESTINATION lib${LIB_SUFFIX}
 				)
 
-# Static library
-if ( STATICLIBS )
-	add_library(ftdipp1-static STATIC ${cpp_sources})
-	set_target_properties(ftdipp1-static PROPERTIES OUTPUT_NAME "ftdipp1")
-	set_target_properties(ftdipp1-static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
-
-	install ( TARGETS ftdipp1-static
-						ARCHIVE DESTINATION lib${LIB_SUFFIX}
-						COMPONENT staticlibs
-					)
-endif ()
-
 install ( FILES ${cpp_headers}
 					DESTINATION include/${PROJECT_NAME}
 					COMPONENT headers
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 17b3617..ae4fc61 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -21,7 +21,7 @@ configure_file(ftdi_version_i.h.in "${CMAKE_CURRENT_BINARY_DIR}/ftdi_version_i.h
 set(c_sources     ${CMAKE_CURRENT_SOURCE_DIR}/ftdi.c ${CMAKE_CURRENT_SOURCE_DIR}/ftdi_stream.c CACHE INTERNAL "List of c sources" )
 set(c_headers     ${CMAKE_CURRENT_SOURCE_DIR}/ftdi.h CACHE INTERNAL "List of c headers" )
 
-add_library(ftdi1 SHARED ${c_sources})
+add_library(ftdi1 ${c_sources})
 
 math(EXPR VERSION_FIXUP "${MAJOR_VERSION} + 1")    # Compatibility with previous releases
 set_target_properties(ftdi1 PROPERTIES VERSION ${VERSION_FIXUP}.${MINOR_VERSION}.0 SOVERSION 2)
@@ -38,17 +38,6 @@ install ( TARGETS ftdi1
           ARCHIVE DESTINATION lib${LIB_SUFFIX}
         )
 
-if ( STATICLIBS )
-  add_library(ftdi1-static STATIC ${c_sources})
-  target_link_libraries(ftdi1-static ${LIBUSB_LIBRARIES})
-  set_target_properties(ftdi1-static PROPERTIES OUTPUT_NAME "ftdi1")
-  set_target_properties(ftdi1-static PROPERTIES CLEAN_DIRECT_OUTPUT 1)
-  install ( TARGETS ftdi1-static
-            ARCHIVE DESTINATION lib${LIB_SUFFIX}
-            COMPONENT staticlibs
-          )
-endif ()
-
 install ( FILES ${c_headers}
           DESTINATION include/${PROJECT_NAME}
           COMPONENT headers
-- 
2.27.0

